tenzoMath/fanucModel.cpp

```diff
@@ -0,0 +1,326 @@
+                    b * (h * cos(theta.at<double>(it, 2)) + i * sin(theta.at<double>(it, 2) + j))) / (a * g - f * b);
+        sintheta = (f * (c * cos(theta.at<double>(it, 2)) + d * sin(theta.at<double>(it, 2)) + e) -
+                    a * (h * cos(theta.at<double>(it, 2)) + i * sin(theta.at<double>(it, 2) + j))) / (a * g - f * b);
+        if (sintheta >= 0)
+            theta.at<double>(it, 0) = acos(costheta);
+        else
+            theta.at<double>(it, 0) = -acos(costheta);
+    }
+
+    for (int it = 0; it < numberOfRoots; ++it)
+    {
+        const double a11 = param[1]._aParam + param[2]._aParam * cos(theta.at<double>(it, 2)) +
+                           param[3]._dParam * sin(param[2]._alphaParam) * sin(theta.at<double>(it, 2));
+        const double a12 = -param[2]._aParam * cos(param[1]._alphaParam) *
+                           sin(theta.at<double>(it, 2)) + param[2]._dParam * sin(param[1]._alphaParam) + param[3]._dParam *
+                           sin(param[2]._alphaParam) * cos(param[1]._alphaParam) * cos(theta.at<double>(it, 2)) +
+                           param[3]._dParam * sin(param[1]._alphaParam) * cos(param[2]._alphaParam);
+        costheta = (a11 * (coord[0] * cos(theta.at<double>(it, 0)) + coord[1] * sin(theta.at<double>(it, 0)) - param[0]._aParam)
+                    - a12 * (-coord[0] * cos(param[0]._alphaParam) * sin(theta.at<double>(it, 0)) + coord[1] *
+                             cos(param[0]._alphaParam) * cos(theta.at<double>(it, 0)) + (coord[2] - param[0]._dParam) * sin(param[0]._alphaParam)))
+                   / (a11 * a11 + a12 * a12);
+        sintheta = (a12 * (coord[0] * cos(theta.at<double>(it, 0)) + coord[1] * sin(theta.at<double>(it, 0)) - param[0]._aParam) + a11 *
+                    (-coord[0] * cos(param[0]._alphaParam) * sin(theta.at<double>(it, 0)) + coord[1] *
+                     cos(param[0]._alphaParam) *
+                     cos(theta.at<double>(it, 0)) + (coord[2] - param[0]._dParam) * sin(param[0]._alphaParam))) / (a11 * a11 + a12 * a12);
+        if (sintheta >= 0)
+            theta.at<double>(it, 1) = acos(costheta);
+        else
+            theta.at<double>(it, 1) = -acos(costheta);
+    }
+    for (int it = 0; it < numberOfRoots; ++it)
+    {
+        theta.at<double>(it, 1) = -theta.at<double>(it, 1) + PI / 2;
+        theta.at<double>(it, 2) -= theta.at<double>(it, 1);
+    }
+
+    std::vector<int> ind;
+
+    for (int it = 0; it < numberOfRoots; ++it)
+    {
+        bool isOk = true;
+
+        if (abs(theta.at<double>(it, 0)) > 170. / 180. * PI)
+            isOk = false;
+        if (theta.at<double>(it, 1) > 90. / 180. * PI || theta.at<double>(it, 1) < -70 / 180. * PI)
+            isOk = false;
+        if (theta.at<double>(it, 2) > 200. / 180. * PI || theta.at<double>(it, 2) < -70 / 180. * PI)
+            isOk = false;
+
+        if (!std::isnan(theta.at<double>(it, 1)) && isOk)
+            ind.emplace_back(it);
```

Comment:
(**optional**): For trivial contructable types move constructor == copy constructor. That's mean you don't need to use on them `std::move()` or such method as `emplase_back()`. It's redundant and decrease readability.